### PR TITLE
added an endpoint for articles (GET)

### PR DESCRIPTION
**Pull request by tartebabba**
### Details - Changelog/Changes:

* Added an endpoint for articles, displaying articles by ID
* Re-factored tests per feedback: assert types for error, provided meaning key for topics, and changed fs requirement in integrations.test.

### Impact: 

* Provides a view per article.

### Testing:

* Addition of 200 for endpoint.
* Checked the JSON received is the JSON expected.

Further notes: Forgot to remove the comments, will be removed in next PR.

### CODE REVIEW

File: __tests__/integrations.test.js [hunk 1]
```diff
@@ -22,10 +23,11 @@ describe('404: Incorrect URL', () => {
       .get('/api/invalid_url')
       .expect(404)
       .then(({ body }) => {
-        const expectedErrorMessage = `Oops.. We can't find that page for you`;
-        const expectedErrorParagraph = `We've searched high and low, but couldn't find the page you're looking for. How about we start from home again?`;
-        expect(body.errorMessage).toBe(expectedErrorMessage);
-        expect(body.errorParagraph).toBe(expectedErrorParagraph);
+        const expectedResponse = {
+          errorMessage: `Oops.. We can't find that page for you`,
+          errorParagraph: `We've searched high and low, but couldn't find the page you're looking for. How about we start from home again?`,
```
Comment (roisinoreilly) [inline __tests__/integrations.test.js:28]: this is maaaybe too much detail for a backend error, but absolutely perfect for when we go onto frontend and thinking about frontend error handling

File: __tests__/integrations.test.js [hunk 2]
```diff
@@ -53,11 +53,67 @@ describe('Topics', () => {
...
+          'created_at',
+          'votes',
+          'article_img_url',
+        ];
+        expectedKeys.forEach((key) => {
+          expect(key in article).toBe(true);
+        });
+        expect(Object.keys(article).length).toBe(8);
+        expect(article).toMatchObject(expectedArticle);
+        expect(typeof article.article_id).toBe('number');
```
Comment (roisinoreilly) [inline __tests__/integrations.test.js:103]: a better jest matcher for testing objects, especially as you'll start testing bigger ones in the next few tickets, is .toMatchObject or .objectContaining. I'd suggest having a go at using them on future tickets

File: __tests__/integrations.test.js [hunk 3]
```diff
@@ -53,11 +53,67 @@ describe('Topics', () => {
...
+        expect(typeof article.author).toBe('string');
+        expect(typeof article.body).toBe('string');
+        expect(typeof article.created_at).toBe('string');
+        expect(typeof article.votes).toBe('number');
+        expect(typeof article.article_img_url).toBe('string');
+      });
+  });
+});
+
+// [x] Update fetch-topics to send back with meaningful key.
```
Comment (roisinoreilly) [inline __tests__/integrations.test.js:115]: I would remove any of these comments before pushing up finished code

**Review by roisinoreilly — COMMENTED** [review]
nice work so far. You're just missing any error handling for this ticket which you do need, I'd like at least 400 and 404 tests please :)

File: __tests__/integrations.test.js (hunk 3 above)
Comment (tartebabba) [reply __tests__/integrations.test.js:115]: Ah, you commented on the bit that caused my issue with the branch! It was these comments which I was trying to resolve - all clear now :)

File: __tests__/integrations.test.js (hunk 1 above)
Comment (tartebabba) [reply __tests__/integrations.test.js:28]: I had the front-end in mind when choosing the errorMessage et paragraph.

**Comment by tartebabba** [comment]
The error handling should be fully there now in the new branch

**Pull request merged**